lms/djangoapps/edraak_university/views.py

```diff
@@ -153,6 +224,21 @@ def get_success_url(self):
             'course_id': self.get_course_key(),
         })
 
+    def delete(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        email = self.object.get_email()
+        try:
+            membership_cohort_id = CohortMembership.objects.get(user=self.object.user, course_id = self.object.course_key).course_user_group.id
+            cohort = CourseUserGroup.objects.get(id=membership_cohort_id)
+            try:
+                remove_user_from_cohort(cohort, email)
```

Comment:
This raises ValueError if course_key is invalid, which is a corner case that is already addressed in the `ensure_valid_course_key` decorator on the dispatch method below.